### PR TITLE
Detect CIDRs vs hardcoding for Helm deploys

### PR DESCRIPTION
**Pull request by dfarrell07**
Use subctl's networking information tooling to dynamically detect
Cluster/Service CIDRs required by Helm. Avoids a fragile dependency on
the internal details of the deployment logic.

Closes: #372

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

**Comment by dfarrell07** [comment]
Per recent discussions, I should probably swap the order of these commits to make this fix able to go in before the main adding-Armada commit, as that might not go in anytime soon (if at all).

**Comment by mangelajo** [comment]
There's an armada patch in the middle

File: scripts/kind-e2e/e2e.sh [hunk 1]
```diff
@@ -201,6 +201,23 @@ function del_subm_gateway_label() {
...
+}
+
+function travis_retry() {
+    # We don't pretend to support commands with multiple words
+    $1 || (sleep 2 && $1) || (sleep 10 && $1)
+}
+
+function install_subctl() {
+    test -x /go/bin/subctl && return
+    version=$(travis_retry get_latest_subctl_tag || echo v0.1.0)
```
Comment (skitt) [inline scripts/kind-e2e/e2e.sh]: Might as well bump to 0.1.1 now ;-).

Comment (dfarrell07) [reply scripts/kind-e2e/e2e.sh]: Done, thanks:)

File: scripts/kind-e2e/lib_helm_deploy_subm.sh [hunk 2]
```diff
@@ -54,12 +55,33 @@ function setup_broker() {
...
+    echo $service_cidr
+}
+
 function helm_install_subm() {
     cluster_id=$1
     crd_create=$2
-    cluster_cidr=$3
-    service_cidr=$4
-    global_cidr=$5
+    global_cidr=$3
```
Comment (sridhargaddam) [inline scripts/kind-e2e/lib_helm_deploy_subm.sh]: Thanks for fixing it, the CI logs look good.

**Comment by dfarrell07** [comment]
> I'd like to question the necessity of this change in light of the decision to stay with bash scripts.
> 
> To me this adds a lot of complexity which doesn't give any tangible value since we're not going forward with armada.

Yeah as I think about it I guess you're right, this isn't worth the complexity anymore.

**Pull request closed without merging**